### PR TITLE
Django4 Compatibility

### PR DESCRIPTION
**Pull request by fleur**
This PR:

- fixes most of the flake8 errors & ignores line too long & line break before and after binary operator
- expands tox.ini to handle django up to 4.1 & python up to 3.10
- 3 commits to make django-mailbox compatible with django4
- revamps tox.ini to add github ci
- fixes first 3 badges and test status link at bottom (please note the build badge will not work until a github ci run has completed)
- fixed a couple deprecation messages seen in tox output

Since the build badge didn't link to anything current, I assumed the travis-ci integration was defunct.  I realized after I did the github integration work that there was a PR in 2020 to fix it.  Please let me know if you feel I should take out the github ci integration.

And from the list in #249:

- updates classifiers in setup.py
- updates CI to tests on Django 4.0, as mentioned above
- updates docs to mark Django 4.0 compatibility
- verifies if there is any pending migrations (answer: no)
- please note does NOT update release note to mark Django 4.0
- fixes any failing tests after CI update, as mentioned above

Also updated docs to remove travis-ci mentions.

### CODE REVIEW

**Comment by t-book** [comment]
@coddingtonbear Why not merge for django 4 compatibility?

**Comment by coddingtonbear** [comment]
Could you, @t-book , and at least one other person, take the time to review this PR for errors and manually test that these changes do functionally work?  A few things to check would include, but not be limited to:

- Interactions in the admin.
- Testing handling for different kinds of incoming mail handlers via the management command.

File: tox.ini [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
-    py{35,36}-django111
-    py{35,36}-django20
-    py{35,36,37}-django21
-    py{35,36,37}-django22
-    py{36,37,38}-django30
+    py35-django{111,20,21,22}
+    py36-django{111,20,21,22,30,31,32}
+    py37-django{21,22,30,31,32}
+    py38-django{30,31,32,40,41}
+    py39-django{30,31,32,40,41}
```
Comment (pfouque) [inline tox.ini:17]:
```suggestion
    py38-django{30,31,32,40,41, 42}
    py39-django{30,31,32,40,41, 42}
    py310-django{30,31,32,40,41, 42}
    py311-django{30,31,32,40,41, 42}
    py312-django{30,31,32,40,41, 42}
```

File: tox.ini [hunk 2]
```diff
@@ -15,7 +23,11 @@ deps=
     django20: django==2.0.*
     django21: django==2.0.*
     django22: django==2.0.*
-    django30: django==3.0b1
+    django30: django==3.0.*
+    django31: django==3.1.*
+    django32: django==3.2.*
+    django40: django==4.0.7
+    django41: django==4.1
```
Comment (pfouque) [inline tox.ini:30]: Do we need to keep deprecated django versions here?
```suggestion
    django41: django==4.1
    django42: django==4.2
```

File: rtd_requirements.txt [hunk 3]
```diff
@@ -1,2 +1,2 @@
-django>=3.1.1,<3.2
+django>=3.1.1,<4.2
```
Comment (pfouque) [inline rtd_requirements.txt:1]:
```suggestion
django>=3.1.1,<5.0
```

**Review by pfouque — APPROVED** [review]
@coddingtonbear , it LGTM!
Since the PR is one year old, it requires an update but apart from that everything looks fine!

Thanks @fleur

**Comment by Pietro395** [comment]
@coddingtonbear Could you merge this PR into a development branch so as to also have the recent changes regarding Office365?
We are ready to test it but we also need Office365 implementation

Thank you

**Review by ptav — APPROVED** [review]
Have tested successfully, including o365 functionality.

**Comment by Pietro395** [comment]
> Have tested successfully, including o365 functionality. 

Hi prav, se have released the 4.9.0 version with django 4 and O365 support

**Review by ptav — APPROVED** [review]
have tested after merging with master. Code is being use in production on our intranet webapp

**Pull request closed without merging**